backend/estimate-service/src/main/java/org/egov/service/NotificationService.java

```diff
@@ -0,0 +1,262 @@
+
+        log.info("build Message For Reject Action");
+        message = buildMessageForRejectAction(estimate, smsDetails, message);
+        SMSRequest smsRequest = SMSRequest.builder().mobileNumber(smsDetails.get("mobileNumber")).message(message).build();
+
+        log.info("push message for REJECT Action");
+        producer.push(config.getSmsNotifTopic(), smsRequest);
+    }
+
+    private void pushNotificationToCreatorForApproveAction(EstimateRequest request) {
+        Estimate estimate = request.getEstimate();
+        String createdByUuid = request.getEstimate().getAuditDetails().getCreatedBy();
+
+        log.info("get message template of creator for approve action");
+        String message = getMessage(request);
+
+        if (StringUtils.isEmpty(message)) {
+            log.info("SMS content has not been configured for this case");
+            return;
+        }
+
+        //get project number, location, userDetails
+        log.info("get project number, location, userDetails");
+        Map<String, String> smsDetails = getDetailsForSMS(request, createdByUuid);
+
+        log.info("build Message For Approve Action for Estimate Creator");
+        message = buildMessageForApproveAction_Creator(estimate, smsDetails, message);
+        SMSRequest smsRequest = SMSRequest.builder().mobileNumber(smsDetails.get("mobileNumber")).message(message).build();
+
+        log.info("push Message For Approve Action for Estimate Creator");
+        producer.push(config.getSmsNotifTopic(), smsRequest);
+    }
+
+    private Map<String, String> getDetailsForSMS(EstimateRequest request, String uuid) {
+        RequestInfo requestInfo = request.getRequestInfo();
+        Estimate estimate = request.getEstimate();
+        String tenantId = estimate.getTenantId();
+
+        Map<String, String> smsDetails = new HashMap<>();
+
+
+        //fetch the logged in user's mobileNumber, username, designation
+        Map<String, String> userDetailsForSMS = hrmsUtils.
+                getEmployeeDetailsByUuid(requestInfo, tenantId, uuid);
+
+
+        // fetch project details - project name and location
+        Map<String, String> projectDetails = getProjectDetails(request);
+
+
+        //get location name from boundary type
+        String boundaryCode = projectDetails.get("boundary");
+        Map<String, String> locationName = locationServiceUtil.getLocationName(tenantId, requestInfo, boundaryCode);
+
+        smsDetails.putAll(userDetailsForSMS);
+        smsDetails.putAll(projectDetails);
+        smsDetails.putAll(locationName);
+
+        return smsDetails;
+    }
+
+    private Map<String, String> getProjectDetails(EstimateRequest request) {
+        Object projectRes = projectServiceUtil.getProjectDetails(request);
+
+        Map<String, String> projectDetails = new HashMap<>();
+        List<String> projectNames = new ArrayList<>();
+        List<String> boundaries = new ArrayList<>();
+        List<String> boundaryTypes = new ArrayList<>();
```

Comment:
List projectNames, boundaries and boundaryTypes can be declared here only. I think no need to initialise here

Reply:
Is this a breaking change? I just wanted to be as explicit as possible so code reading becomes easier